[CORL-462] Update the wordpress plugin to support Coral version 5

nick-funk opened the pull request:
This updates the plugin to allow switching between version 4 and 5.

Instructions for how to setup a local development environment to test this is available at:

https://app.tettra.co/teams/coralproject/pages/wordpress-development-environment

nick-funk committed:
Allow users to select their talk version and embed v5 comments

CORL-462

nick-funk committed:
Replace all displayed text instances of `Talk` with `Coral` or similar.

CORL-462

nick-funk commented on README.md:
```diff
@@ -40,6 +40,8 @@ if ( comments_open() ) {
-Talk version >= `4.0.0` use talk-wp-plugin `v0.1.0`
+Coral version >= `4.0.0` use plugin version `v0.1.0`
+
+Coral version >= `5.0.0` use plugin version `v0.2.0`
```
@wyattjoh, I've indicated that there is a version bump for the plugin to support the v4/5 switching in this PR.

Please advise on what version number we want to use so I can update it in source as well as package this up and/or tag the version on master appropriately.

wyattjoh replied:
v0.2.0 is good as this isn't a breaking change to existing v4 installs as far as I know.

nick-funk commented on inc/comments-template.php:
```diff
@@ -11,12 +11,38 @@
+			Coral.createStreamEmbed({
+				id: "coral_thread",
+				autoRender: true,
+				rootURL: "<?php echo esc_url( $talk_url ); ?>",
```
@wyattjoh, we have a nice embed here that seems to work with my local dev environment.

Note that I haven't specified `storyID` or `storyURL`. Is this fine, or should I pull them from somewhere? Perhaps convenience functions are available in the PHP scripts that I'm not aware of?

wyattjoh replied:
You could add [`get_the_ID()`](https://developer.wordpress.org/reference/functions/get_the_id/) to this as the `storyID` if you want I suppose!

nick-funk commented on inc/talk-settings-static-content.php:
```diff
@@ -8,20 +8,20 @@
 	<?php printf(
-		esc_html__( 'You can find out how to install and manage Talk %shere%s.', 'coral-project-talk' ),
+		esc_html__( 'You can find out how to install and manage Coral %shere%s.', 'coral-project-talk' ),
 		'<a href="https://docs.coralproject.net/talk/">',
```
Changing up the names to `Coral` everywhere when the URL still points to `...something coral.../talk/` seems a bit weird.

Not sure if we want to fully go forward with the change of name to Coral with this plugin release, or wait until we've consolidated all document with our next release?

wyattjoh replied:
I'd just leave this for now until we get the docs handled, then it'll have to be managed across the board.

wyattjoh commented on inc/comments-template.php:
```diff
@@ -11,12 +11,38 @@
+	exit();
+endif;
+
+if ( $talk_version == "4" ) : ?>
```
For fallback, you should instead do something like `if ( $talk_version ~= "5" ) : ?>` instead, as those who have the plugin installed already should still load the correct template.

nick-funk replied:
Excellent idea and good catch, will update shortly!

nick-funk committed:
Set the storyID when embedding v5 comment streams

Allows Coral to properly distinguish which post we're coming from.

Without this, the Stories area will accept new posts as stories, and they
will be distinct, but all named the same as the first post ever created
(very confusing to moderators).

CORL-462

nick-funk committed:
When embedding the comments, fall back to v4 if Coral version is not v5

This is due to the fact that v4 covers all previous versions of comments
and not just v4.

CORL-462

nick-funk committed:
Bump plugin version 0.2.0 due to new v5 support changes

CORL-462

nick-funk committed:
Merge remote-tracking branch 'origin/master' into feature/CORL-462

# Conflicts:
#	talk.php

wyattjoh commented on inc/class-talk-settings-page.php:
```diff
@@ -145,6 +154,40 @@ public function render_container_classes_field() {
+			<option value="5"
+				<?php 
+					if (esc_attr( get_option( 'coral_talk_version' ) ) === "5")
+						echo "selected=\"selected\""
```
I would suggest moving the `esc_attr( get_option( 'coral_talk_version' ) )` to the top of the `render_version_field` function if we can, then it should be good to merge!

wyattjoh commented on inc/comments-template.php:
```diff
@@ -11,8 +11,34 @@
+	</script>
+<?php endif;
+
+if ( $talk_version != "5" ) : ?>
```
Why not `else`?

nick-funk replied:
Wanted to explain what "else" would be, in this case, everything prior to 5.
Will update and add comment.

nick-funk committed:
Pull talk version retrieval into a re-usable variable

CORL-462

nick-funk committed:
Simplify logic for displaying v4 or v5 comment sections on posts

CORL-462
